src/Raven.Server/Config/Categories/SecurityConfiguration.cs

```diff
@@ -1,63 +1,143 @@
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.MasterKey:Path")]
+        [ConfigurationEntry("Security.MasterKey.Path")]
         public string MasterKeyPath { get; set; }
 
         [Description("A command or executable to run which will provide a (512-bit) Master Key, If specified, RavenDB will use this key to protect secrets.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.MasterKey:Exec")]
+        [ConfigurationEntry("Security.MasterKey.Exec")]
         public string MasterKeyExec { get; set; }
 
         [Description("The command line arguments for the 'Security.MasterKey:Exec' command or executable.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.MasterKey:Arguments")]
+        [ConfigurationEntry("Security.MasterKey.Arguments")]
         public string MasterKeyExecArguments { get; set; }
 
         [Description("The number of milliseconds to wait for the Master Key executable to exit. Default: 30 seconds")]
         [DefaultValue(30000)]
-        [ConfigurationEntry("Security.MasterKey:Timeout")]
+        [ConfigurationEntry("Security.MasterKey.Timeout")]
         public int MasterKeyExecTimeout { get; set; }
+
+        [Description("If authentication is disabled, set address range type for which server access is unsecured (None | Local | PrivateNetwork | PublicNetwork).")]
+        [DefaultValue(nameof(UnsecuredAccessAddressRange.Local))]
+        [ConfigurationEntry("Security.Authentication.UnsecuredAccessAllowed")]
+        public FlagSetting<UnsecuredAccessAddressRange> UnsecuredAccessAllowed { get; set; }
+
+        public UnsecuredAccessAddressRange UnsecuredAccessAddressRange => UnsecuredAccessAllowed.SettingValue.Value;
+
+        internal bool? IsUnsecureAccessSetupValid { get; private set; }
+
+        internal string UnsecureAccessWarningMessage { get; private set; }
+
+        public bool IsCertificateConfigured => string.IsNullOrWhiteSpace(CertificatePath) == false 
+            || string.IsNullOrWhiteSpace(CertificateExec) == false;
+
+        public bool AuthenticationEnabled => IsCertificateConfigured;
+
+        internal static void Validate(RavenConfiguration configuration)
+        {
+            var serverUrl = configuration.Core.ServerUrl.ToLowerInvariant();
+            if (Uri.TryCreate(serverUrl, UriKind.Absolute, out var uri) == false)
+                throw new UriFormatException("Unable to parse URL - " + serverUrl);
+
+            var isServerUrlHttps = uri.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase);
+
+            var serverAddresses = DetermineServerIp(uri);
+            var unsecuredAccessAddressRange = configuration.Security.UnsecuredAccessAddressRange;
+            var serverIsWithinUnsecuredAccessRange = serverAddresses.Any(x => SecurityUtils.IsUnsecuredAccessAllowedForAddress(unsecuredAccessAddressRange, x));
+
+            if (configuration.Security.AuthenticationEnabled)
```

Comment:
configuration.Security.AuthenticationEnabled is just a getter checking certs are there. Having a cert set up == authentication enabled.